dev_handler/Makefile

```diff
@@ -1,10 +1,30 @@
-LIBFLAGS=-pthread -lrt -Wall
-BIN=../bin
+# list of libraries that dev_handler needs to compile
+LIBS=-pthread -lrt -Wall
 
-dev_handler: dev_handler.c message.c message.h ../logger/logger.c ../runtime_util/runtime_util.c ../shm_wrapper/shm_wrapper.c
-	gcc $^ -o $(BIN)/dev_handler $(LIBFLAGS)
+# list of source files that the target (dev_handler) depends on, relative to this folder
+SRCS = dev_handler.c dev_handler_message.c ../logger/logger.c ../runtime_util/runtime_util.c ../shm_wrapper/shm_wrapper.c
 
+# specify the target (executable we want to make)
+TARGET = dev_handler
+
+.PHONY: all clean $(TARGET)
```

Comment:
Ben: "basically it tells make that the thing listed after .PHONY is not a real file; in this case we don’t actually make a file called all, clean, or dev_handler (we make a file called ../bin/dev_handler)
basically it prevents make from redo-ing commands, like say we want to have make dev_handler  build ../bin/dev_handler. if we don’t declare dev_handler as .PHONY then when you run make dev_handler it will always try to look for a file called dev_handler, which will never exist (cuz really the file is ../bin/dev_handler ) so it will run the compilation command again when it doesn’t need to"

Adding it here for future reference.

Reply:
Just to clean up this message a bit since that was something I hastily wrote in Slack:

If you're completely confused, read our wiki page on `make` [here](https://github.com/pioneers/runtime/wiki/Make).

But in short, this Makefile contains the targets `all`, `clean`, and `dev_handler`. However, none of these targets are real files (there is no file at `runtime/dev_handler/all`, `runtime/dev_handler/clean`, or `runtime/dev_handler/dev_handler`). That last one might be a bit confusing--the `dev_handler` executable is located in `runtime/bin/dev_handler`, so from the perspective of the directory that this Makefile is in, the path is `../bin/dev_handler`.

Thus, if we just left the Makefile with these three targets, each time we ran `make all` or `make clean` or `make dev_handler`, it would try to find a file called `all`, `clean`, or `dev_handler`, respectively, fail to find the fail (since we never make a file with that name), and *always* run the recipe (which is not what we want to do!). We only want to run the recipe if there is something to be made.

Thus, we tell `make` that these three special targets are not real files by labeling them as `.PHONY`. Thus, when we do `make dev_handler`, it will it won't automatically build everything just because it can't find the `dev_handler` file (which will never exist).